src/main/java/portfmgr/view/PortfolioDetailViewController.java

```diff
@@ -1,15 +1,31 @@
 package portfmgr.view;
 
 import javafx.fxml.FXML;
+
 import javafx.fxml.FXMLLoader;
 import javafx.scene.control.Button;
 import javafx.scene.layout.AnchorPane;
 import portfmgr.portfmgrApplication;
+import portfmgr.model.Portfolio;
+/**
+ * This class represents one of the main controllers and manage the detail view about the portfolios.
+ * 
+ * @author Marc Steiner
+ */
 
 public class PortfolioDetailViewController {
 
-	// Reference to the main app (portfmgrApplication)
 	private portfmgrApplication mainApp;
+	private Portfolio portfolio;
+	private String portfolioName;
+	private String portfolioCurrency;
+	
+	public PortfolioDetailViewController(Portfolio portfolio) {
+		this.portfolio = portfolio;
+		portfolioName = portfolio.getPortfolioName();
+		portfolioCurrency = portfolio.getPortfolioCurrency();
+		updatePortfolio();
+	}
```

Comment:
I don't see the reason for the constructor. Does it really need it? If so, for what?

Reply:
No constructor added